subprojects/integ-test/src/crossVersionTest/groovy/org/gradle/integtests/TaskSubclassingBinaryCompatibilityCrossVersionSpec.groovy

```diff
@@ -121,35 +121,41 @@ class TaskSubclassingBinaryCompatibilityCrossVersionSpec extends CrossVersionInt
             }
         """
 
+        // Since Gradle 8.0 all task classes are abstract, and their injected properties can also be abstract methods
+        def abstractness = previous.version.baseVersion >= GradleVersion.version("8.0") ? "abstract" : ""
```

Comment:
Did you check if we could always make classes abstract?

Reply:
I haven't, but I don't think we want to make the sub-class in the test abstract unconditionally.

If a user subclassed a task before Gradle 8.0, this sub-class could have legitimately been non-abstract. In this test we want to make sure this stays true, in case this is suddenly gets broken in something like 7.6.1.